### PR TITLE
[recnet-api] Self-rec flag 

### PR DESCRIPTION
**Pull request by swh00tw**
## Description

We need a “self-rec” flag to display at feeds and email digest. 

This PR updated prisma schema and update APIs to enable marking rec as "self-rec".

Design docs: https://www.notion.so/Self-review-feature-4fcb73e72d0d41abbb4dafef0c672512?pvs=4

## Related Issue

- #269 

## Test

- Open recnet web app and try to create rec/edit rec.
- All bahaviors should remain the same.
- Try to use Postman to turn a rec to “self-rec” rec
- Verify the API is working properly via inspecting db (open prisma studio)
- Try to use Postman to create a “self-rec” rec
- Verify the API is working properly via inspecting db (open prisma studio)

## TODO

- [x] Paste the testing link
- [x] Clear `console.log` or `console.error` for debug usage
- [x] Update the documentation `recnet-docs` if needed
- [x] Version bump in `package.json` if needed

### CODE REVIEW

File: apps/recnet-api/prisma/schema.prisma
Comment (swh00tw) [inline apps/recnet-api/prisma/schema.prisma:1]: Lots of changes due to formatter. We should try to configure prettier or formatter properly for `*.prisma` file or different IDE will have different formatting and we will keep seeing this. Not a critical thing here, we will fix in other PR.

Comment (swh00tw) [inline apps/recnet-api/prisma/schema.prisma:1]: I guess we should configure formatter properly for `*.prisma` file in other PR or we will keep seeing lots of change everytime we make changes to `schema.prisma`. Not a critical issue here tho.

**Comment by swh00tw** [comment]
Please ignore the changes in `recnet` and `recnet-docs`, I guess they are there since I created this branch from `main` but the branch we are going to merge into is `recnet-api-dev`. I did rebase `recnet-api-dev` tho, not sure why those changes are still there. 🤔

**Review by joannechen1223 — APPROVED** [review]
LGTM

**Pull request merged**